controller/controller_server/exception_handler.py

```diff
@@ -0,0 +1,43 @@
+from decorator import decorator
+
+import grpc
+
+import controller.array_action.errors as controller_errors
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ObjectIdError
+from controller.controller_server.errors import ValidationException
+from controller.csi_general import csi_pb2
+
+logger = get_stdout_logger()
+
+exceptions_to_status_codes_map = {
+    ValidationException: grpc.StatusCode.INVALID_ARGUMENT,
+    ObjectIdError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.IllegalObjectName: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.SpaceEfficiencyNotSupported: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotExist: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolDoesNotMatchCapabilities: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.PoolParameterIsMissing: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ExpectedSnapshotButFoundVolumeError: grpc.StatusCode.INVALID_ARGUMENT,
+    controller_errors.ObjectNotFoundError: grpc.StatusCode.NOT_FOUND,
+    controller_errors.PermissionDeniedError: grpc.StatusCode.PERMISSION_DENIED,
+    controller_errors.VolumeAlreadyExists: grpc.StatusCode.ALREADY_EXISTS,
+    controller_errors.NotEnoughSpaceInPool: grpc.StatusCode.RESOURCE_EXHAUSTED
```

Comment:
please leave here only the pairs in which the exception *always* leads to the status code.
e.g. `ObjectIdError` does not belong here, since it doesn't always lead to `NOT_FOUND`

Reply:
Done

Reply:
please also remove uncommon exceptions, e.g. `SpaceEfficiencyNotSupported`, since it is method-specific

Reply:
another way to reduce `except` clauses is to make same-status-codes exceptions inherit from the same intermediate class

Reply:
What do you mean by "method-specific"?
I would like the program to handle exceptions as much as it can instead of handling them in the server.

Reply:
I mean that the exception is relevant for only one method.
the purpose of this decorator is to have common handlings in one place, not to have as much handlings as it can.
the default place for exception handling should still be in the server, unless it already exists somewhere else there, in which case it can move to the common decorator.

Reply:
Done

Reply:
I still see some, e.g. [`SpaceEfficiencyNotSupported`](https://github.com/IBM/ibm-block-csi-driver/pull/305#discussion_r608638835)...

Reply:
Done